compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirDeclarationsResolveTransformer.kt

```diff
@@ -707,7 +707,12 @@ open class FirDeclarationsResolveTransformer(transformer: FirBodyResolveTransfor
         if (valueParameter.returnTypeRef is FirImplicitTypeRef) {
             transformer.replaceDeclarationResolvePhaseIfNeeded(valueParameter, transformerPhase)
             valueParameter.replaceReturnTypeRef(
-                valueParameter.returnTypeRef.errorTypeFromPrototype(ConeSimpleDiagnostic("Unresolved value parameter type"))
+                valueParameter.returnTypeRef.errorTypeFromPrototype(
+                    ConeSimpleDiagnostic(
+                        "No type for parameter",
+                        DiagnosticKind.ValueParameterWithNoTypeAnnotation
```

Comment:
Nice catch! I wasn't aware of this diagnostic when I added: https://github.com/JetBrains/kotlin/commit/9aaa952b39b0e7882e7eb78b3ed981ed7a84ddac